build LaTeX in PRs

Shoeboxam opened the pull request:
Closes #331 

- adds a GH actions workflow that
    1. checks out the full repository
    2. uses a git command to find a list of edited .tex files in this PR
    3. if the diff is non-empty, opens a new job to build the .tex files in the diff
    4. the list of diffed .tex files is passed to the latex-action step, which builds them in-place
    5. uploads all .pdfs to a [new artifacts repository](https://github.com/opendp/artifacts)
    6. adds or updates the github-actions message with links to said .pdfs (see below)
- commits are made to the artifacts repo by the dummy OpenDP user account named `opendp-dev`
    - `opendp-dev` only has permissions to the artifacts repo
    - `opendp-dev` has an authentication token in the opendp repository's secrets
- steps 5 and 6 are handled by [this fork of an existing MIT-licensed action](https://github.com/opendp/pull-request-artifacts)


I rebased the ported whitepapers proofs on this PR, so that they would get rendered pdfs as well. Holding off to rebase the newer interns' PRs until I talk with them.

andrewvyrros commented on .github/workflows/latex.yml:
```diff
@@ -0,0 +1,50 @@
+        uses: Shoeboxam/pull-request-artifacts@main
+        with:
+          commit: ${{ github.event.pull_request.head.sha }}
+          glob-pattern: rust/**/*.pdf
```
Doing this with a glob pattern seems brittle, could pick up spurious files. You have the tex files in hand, can't you do this explicitly?

andrewvyrros reviewed (commented):
OK, I've come around to the necessity of putting these in an external repo. 😄

I'm a little concerned about having a custom version of the `pull-request-artifacts` action. I worry about this breaking in the future, and the project needing someone who can debug Javascript to fix things. At the very least, we should put the fork in the opendp org. But ideally, we wouldn't need a custom action.

Also, it's not ideal to have a separate `opendp-shoebot` user, as it's another set of credentials to maintain. Can this run as the original user of the PR?

Shoeboxam commented:
I can't commit as the original user of the PR, because I won't have account tokens to commit on someone else's behalf. I could spoof the commit information to make it look like someone else made the commit, but I still have to authenticate with the bot account. I think that gives us the worst of both worlds- still have another set of credentials, and I'd rather not spoof commit identification.

There were several changes made to the action-
1. glob star matching. I know how to work around this now and could omit this completely
2. preserving the file structure in the artifacts repo. I think there's a high likelihood of path collisions if this change isn't made
3. formatting changes in the PR comment (we could do without this)

I could open a PR to the original repo with the changes for 2.

Shoeboxam commented:
I moved the action into the opendp org, dropped the glob matching, and dropped some unnecessary changes. I also switched the bot account from opendp-shoebot to the existing opendp-dev account (we already use this for pypi releases). 

Given the number of changes to the pull-request-artifacts action, I'm not sure if it will be accepted upstream.
